BaragonData/src/main/java/com/hubspot/baragon/data/BaragonStateDatastore.java

```diff
@@ -89,6 +89,24 @@ public void saveService(BaragonService service) {
     writeToZk(servicePath, service);
   }
 
+  public boolean isServiceUnchanged(BaragonRequest update) {
+    if (update.isUpstreamUpdateOnly()) {
+      return true;
+    }
+
+    Optional<BaragonService> maybeExistingService = getService(update.getLoadBalancerService().getServiceId());
```

Comment:
Do we have this data in memory anywhere already? If not this is gonna be an extra zk call on every update